knowledge_engine/knowledge_engine/embedding/factory.py

```
@@ -67,14 +81,38 @@ def _create_embedding_model_from_resolved_values(
+    embed_model,
+    *,
+    additional_input_modalities: list[str],
+):
+    """Expose resolved embedding capabilities on the runtime model instance."""
+    try:
+        object.__setattr__(
+            embed_model,
+            "_additional_input_modalities",
+            list(additional_input_modalities),
+        )
+    except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)